app/src/main/java/org/jellyfin/androidtv/auth/ServerRepository.kt

```diff
@@ -102,8 +102,13 @@ class ServerRepositoryImpl(
 		getServerPublicUsers(server)
 			.sortedBy { it.name }
 			.forEach { user ->
-				if (users.none { it.id == user.id })
-					users.add(user)
+				val index = users.indexOfFirst { it.id == user.id }
+				if (index == -1) users.add(user)
+				else users[index] = (users[index] as PrivateUser).copy(
```

Comment:
Should check if `users[index]` is a PrivateUser so this won't cause issues if the above code is changed in the future.

something like this I guess
```kt
val currentUser = users[index]
if (currentUser is PrivateUser) users[index] = currentUser.copy()
else if (currentUser is PublicUser) users[index] = user
```